### DISCUSSION TITLE
Mutating state in updating hooks

### DISCUSSION THREAD

**Discussion by evanbarter** [livewire/livewire#7870] [1. Help]
Hi, 

I feel like it used to be possible in v2 to mutate the state of a property in an updating hook by making the ```$value``` argument  a reference. In v3 this no longer seems to be possible. Is there any way to accomplish this before the updated hook?

```php
<?php

namespace App\Livewire;

use Livewire\Component;

class Test extends Component
{
    public $test;

    public function updating($name, $value) {
        $this->test = 'abc'; // Nope
    }

    public function updatingTest(&$value) {
        $value = 'def'; // Also nope
        $this->fill(['test' => 'def']); // Still nope
    }

    public function updated($name, $value) {
        logger($this->test);
        logger($value);

        $this->test = 'ghi'; // This will do it
    }

    public function render()
    {
        return <<<'HTML'
        <div>
            <input type="text" wire:model.blur="test">
        </div>
        HTML;
    }
}
```

**Comment by joshhanley** [comment]
@evanbarter hmm I think that should work. Any chance you could submit a failing test? Thanks!

**Comment by thijsvdanker** [comment]
@joshhanley I've added a failing test for this here https://github.com/livewire/livewire/pull/8061

**Reply by calebporzio** [reply]
I commented on the PR, but just to re-iterate: I'm not saying "no". I'm just saying: I'm not going to put in the work on this. I'm also saying: if someone else puts in the work and submits a PR, make sure it's a small, simple PR, otherwise, I likely will not merge it. Thanks again!
